### PR TITLE
better error reporting

### PR DESCRIPTION
**Pull request by h4ck3rm1k3**
Improve upon the error message to include the actual absolute path of the file in question.

### CODE REVIEW

**Comment by thommay** [comment]
LGTM :+1: @chef/client-maintainers

File: lib/chef/api_client/registration.rb [hunk 1]
```diff
@@ -68,7 +68,8 @@ def run
 
       def assert_destination_writable!
         if (File.exists?(destination) && !File.writable?(destination)) or !File.writable?(File.dirname(destination))
-          raise Chef::Exceptions::CannotWritePrivateKey, "I cannot write your private key to #{destination} - check permissions?"
+          abs_path = File.expand_path(destination)
+          raise Chef::Exceptions::CannotWritePrivateKey, "I cannot write your private key to #{destination} (#{abs_path})- check permissions?"
```
Comment (lamont-granquist) [inline lib/chef/api_client/registration.rb]: could probably just replace destination with abs_path here, i don't think the user needs to see both?

Comment (h4ck3rm1k3) [inline lib/chef/api_client/registration.rb]: ok, that is fine

**Comment by h4ck3rm1k3** [comment]
improved conciseness

**Comment by danielsdeleo** [comment]
:+1: The appveyor failures are not related to this patch. There were two run lock failures also, I tried re-running to see if they are intermittent. I know we did change those slightly because the test was triggering a race, but perhaps the fix was not complete enough.

**Pull request closed without merging**